chrome/content/zotero/xpcom/date.js

```diff
@@ -623,14 +623,12 @@ Zotero.Date = new function(){
 		return string;
 	}
 	
-	this.strToISO = function (str) {
-		var date = this.strToDate(str);
-		
-		if(date.year) {
+	this.dateToISO = function (date) {
```

Comment:
Oh, and the tests are failing because `Zotero.Date.dateToISO` actually already exists, expecting a JS Date, and you overrode it, which is causing some code to get `false` instead.